### PR TITLE
Address more exception todos

### PR DESCRIPTION
**Pull request by cocreature**
A bit of a kitchen sink PR to address a bunch of the trivial todos
that didn’t seem worth splitting out into separate PRs.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionSpec.scala [hunk 1]
```diff
@@ -275,9 +273,7 @@ class TransactionSpec extends AnyFreeSpec with Matchers with ScalaCheckDrivenPro
 
   "contractKeys" - {
     "return all the contract keys" in {
-      // TODO: https://github.com/digital-asset/daml/issues/8020
-      // change VDev to  TransactionVersion.StableVersions.max once exception are released
-      val builder = TransactionBuilder(TransactionVersion.VDev)
+      val builder = TransactionBuilder(TransactionVersion.minExceptions)
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionSpec.scala:276]: The test originally works with the latest stable version. 
I don't think we want to change that.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala [hunk 2]
```diff
@@ -244,19 +244,22 @@ class IdeLedgerClient(val compiledPackages: CompiledPackages) extends ScriptLedg
...
+          node match {
+            case create: NodeCreate[ContractId] => ScriptLedgerClient.CreateResult(create.coid)
+            case exercise: NodeExercises[NodeId, ContractId] =>
+              ScriptLedgerClient.ExerciseResult(
+                exercise.templateId,
+                exercise.choiceId,
+                exercise.exerciseResult.get,
+              )
+            case _: NodeFetch[_] | _: NodeLookupByKey[_] | _: NodeRollback[_] =>
+              throw new IllegalArgumentException(s"Invalid root node: $node")
```
Comment (remyhaemmerle-da) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala:261]: :+1:

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionSpec.scala [hunk 3]
```diff
@@ -217,9 +217,7 @@ class TransactionSpec extends AnyFreeSpec with Matchers with ScalaCheckDrivenPro
     "ignores location" in forAll(genEmptyNode) { n =>
       val withoutLocation = {
         val nodeWithoutLocation = n match {
-          case _: Node.NodeRollback[_] =>
-            // TODO https://github.com/digital-asset/daml/issues/8020
-            sys.error("rollback nodes are not supported")
+          case nr: Node.NodeRollback[Nothing] => nr
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionSpec.scala:220]: :+1:

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala [hunk 4]
```diff
@@ -316,8 +316,7 @@ class IdeLedgerClient(val compiledPackages: CompiledPackages) extends ScriptLedg
                   exercise.children.collect(Function.unlift(convEvent(_))).toList,
                 )
               )
-            case _: NodeFetch[ContractId] => None
-            case _: NodeLookupByKey[ContractId] => None
+            case _: NodeFetch[_] | _: NodeLookupByKey[_] | _: NodeRollback[_] => None
```
Comment (remyhaemmerle-da) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala:319]: :+1:

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionSpec.scala (hunk 1 above)
Comment (cocreature) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionSpec.scala]: Reverted in e9a87bb0741c2bfaae8ca23bd616b22a76d7d56c for now.
What are you suggesting we do here in the long term? We clearly cannot go below minExceptions if we use exceptions. Split up tests in some that use exceptions and some that don’t?

**Review by remyhaemmerle-da — APPROVED** [review]
Thanks.

**Comment by remyhaemmerle-da** [comment]
part of #8020

**Pull request merged**